vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -251,6 +252,12 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 	if errAttach != nil {
 		log.WithFields(log.Fields{"name": r.Name,
 			"error": errAttach}).Error("Attach volume failed, removing the volume ")
+		if strings.Contains(errAttach.Error(), recvError) {
```

Comment:
Also, the current code seems a bit too complicated to me. E.g. there are at least 3 error paths with Detach->Remove actions and cut-n-pasted code 

I think what we are trying to do is as follows:
* attach
* if attach fails for any reason, try to detach and try to remove. If either fails. log an error and recommendation on how to fix manually. Abort create()
* mkfs. If it fails, do the "try to detach and try to remove... abort create()" steps (a dedicated function) 
* try to detach, warn if there is a problem
* done